### ISSUE TITLE
4.0-milestone-2: org.gradle.api.tasks.ParallelizableTask removed?

### ISSUE THREAD

**Issue by mkobit** [gradle/gradle#2199]
I was trying to build a project I had using `4.0-milestone-2` and received an `Unresolved reference: ParallelizableTask` in my `buildSrc` tasks which have `@ParallelizableTask` on them.

### Expected Behavior

`ParallelizableTask` should still be present (or document what happened since it was `@Incubating`)

I didn't see any removal notes in https://docs.gradle.org/4.0-milestone-1/release-notes.html or https://docs.gradle.org/4.0-milestone-2/release-notes.html or deprecation notices.

* Javadoc at `3.5` - https://docs.gradle.org/3.5/javadoc/org/gradle/api/tasks/ParallelizableTask.html
* Not at `4.0-milestone-2` - https://docs.gradle.org/4.0-milestone-2/javadoc/org/gradle/api/tasks/ParallelizableTask.html

### Steps to Reproduce (for bugs)

* `buildSrc/build.gradle`

    ```groovy
    plugins {
      id "org.jetbrains.kotlin.jvm" version "1.1.2-2"
    }
    
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8
    
    repositories {
      jcenter()
    }
    
    dependencies {
      compile gradleApi()
      compile "org.jetbrains.kotlin:kotlin-stdlib:1.1.2-2"
      compile("com.github.kittinunf.fuel:fuel:1.3.1")
      testCompile("junit:junit:4.12")
    }
    ```
* Write a task that uses `@ParallelizableTask`

### Your Environment

Can't show too much, but:

```
#Tue May 30 09:47:26 CDT 2017
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.0-milestone-2-bin.zip

```

**Comment by ghale** [comment]
@mkobit `@ParallelizableTask` and intra-project task parallelization were undocumented features that we never officially supported, so we did not go through the normal deprecation cycle before removing them.

To be clear, inter-project task parallelization (i.e. "--parallel") is still supported and works as it always has, it's just the undocumented intra-project parallelization (i.e. use of `@ParallelizableTask` and the magic intra-project parallelism flag) that has changed.  This has been replaced with a new API for running parallelizable work that will allow tasks in the same project to potentially run in parallel.  We plan to document/announce this API in 4.1.

 If you are interested in dog-fooding this API and converting your tasks to use it, we'd love your feedback, but I can't offer much in terms of documentation at this point in time.  I'd have to point you to the code and integration tests for examples.  Let me know if you're interested.
